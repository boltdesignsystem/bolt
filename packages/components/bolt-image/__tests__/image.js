import { render, stopServer, renderWC } from '../../../testing/testing-helpers';
import schema from '../image.schema';
const componentSelector = 'bolt-image';
// eslint-disable-next-line camelcase
const { src, alt, lazyload, max_width, attributes } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    src: '/fixtures/1200x660.jpg',
    alt: 'A Rock Climber',
  };

  const srcData = {
    src: '/fixtures/1200x660.jpg',
  };

  fixtures = {
    defaultData,
    srcData,
  };
});

describe('Bolt Image', () => {
  test(`Image default`, async () => {
    const results = await render('@bolt-components-image/image.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  // @todo: lazysizes returns unpredictable classname order, comment out for now
  // test(`Image default with web component`, async () => {
  //   const results = await render('@bolt-components-image/image.twig', {
  //     ...fixtures.defaultData,
  //   });

  //   const { innerHTML, outerHTML } = await renderWC(
  //     componentSelector,
  //     results.html,
  //     page,
  //   );

  //   await expect(results.ok).toBe(true);
  //   await expect(results.html).toMatchSnapshot();
  //   await expect(innerHTML).toMatchSnapshot();
  //   await expect(outerHTML).toMatchSnapshot();
  // });
});

describe('Bolt Image Props', () => {
  // Target each of the schema keys with the following pattern

  [true, false].forEach(async option => {
    test(`Image ratio object and lazyload:${option} compiles`, async () => {
      const results = await render('@bolt-components-image/image.twig', {
        ...fixtures.defaultData,
        lazyload: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  test(`Image with max-width of 200px renders as expected`, async () => {
    const results = await render('@bolt-components-image/image.twig', {
      ...fixtures.defaultData,
      max_width: '200px',
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Image with max-width of 50% renders as expected`, async () => {
    const results = await render('@bolt-components-image/image.twig', {
      ...fixtures.defaultData,
      max_width: '50%',
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Image with max-width and additional style attribute renders as expected`, async () => {
    const results = await render('@bolt-components-image/image.twig', {
      ...fixtures.defaultData,
      max_width: '200px',
      attributes: {
        style: 'display: block;',
      },
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Image should always render alt attribute on <img> tags`, async () => {
    const results1 = await render('@bolt-components-image/image.twig', {
      ...fixtures.srcData,
      alt: '',
    });

    await expect(results1.ok).toBe(true);
    await expect(results1.html).toMatchSnapshot();

    const results2 = await render('@bolt-components-image/image.twig', {
      ...fixtures.srcData,
    });

    await expect(results2.ok).toBe(true);
    await expect(results2.html).toMatchSnapshot();
  });
});
