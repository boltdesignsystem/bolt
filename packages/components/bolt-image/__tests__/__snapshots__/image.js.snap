// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Image Image default 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%);"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image ratio object and lazyload:false compiles 1`] = `
<bolt-image no-lazy
            ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%);"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           sizes="auto"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image"
           alt="A Rock Climber"
           srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           sizes="auto"
           src="/fixtures/1200x660.jpg"
      >
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image ratio object and lazyload:true compiles 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%);"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image should always render alt attribute on <img> tags 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%);"
            src="/fixtures/1200x660.jpg"
            alt
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image should always render alt attribute on <img> tags 2`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%);"
            src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image with max-width and additional style attribute renders as expected 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="display: block;background-color: hsl(233, 33%, 97%); width: 200px; max-width: 100%;"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
            max-width="200px"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image with max-width of 50% renders as expected 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%); width: 50%;"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
            max-width="50%"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;

exports[`Bolt Image Props Image with max-width of 200px renders as expected 1`] = `
<bolt-image ratio="1200/660"
            placeholder-color="hsl(233, 33%, 97%)"
            placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
            style="background-color: hsl(233, 33%, 97%); width: 200px; max-width: 100%;"
            src="/fixtures/1200x660.jpg"
            alt="A Rock Climber"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
            max-width="200px"
>
  <bolt-ratio ratio="20/11">
    <ssr-keep for="bolt-ratio"
              class="c-bolt-ratio"
              style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
           data-sizes="auto"
           srcset="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
           src="/fixtures/1200x660.jpg"
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w, /fixtures/1200x660.jpg 1200w"
        >
      </noscript>
    </ssr-keep>
  </bolt-ratio>
</bolt-image>
`;
